source/backend/api/Services/DocumentService.cs

```diff
@@ -421,15 +425,24 @@ public async Task<ExternalResponse<string>> DeleteMayanStorageDocumentAsync(long
 
         public async Task<ExternalResponse<QueryResponse<Models.Mayan.Document.DocumentTypeModel>>> GetStorageDocumentTypes(string ordering = "", int? page = null, int? pageSize = null)
         {
-            this.Logger.LogInformation("Retrieving storage document types");
-            this.User.ThrowIfNotAuthorizedOrServiceAccount(Permissions.DocumentView, keycloakOptions);
+            Logger.LogInformation("Retrieving storage document types");
+            User.ThrowIfNotAuthorizedOrServiceAccount(Permissions.DocumentView, keycloakOptions);
 
-            ExternalResponse<QueryResponse<Models.Mayan.Document.DocumentTypeModel>> result = await documentStorageRepository.TryGetDocumentTypesAsync(ordering, page, pageSize);
-            if (result.Status != ExternalResponseStatus.Success)
+            if(!_memoryCache.TryGetValue(CacheKeys.MayanDocumentTypes, out ExternalResponse<QueryResponse<Models.Mayan.Document.DocumentTypeModel>> cachedDocumentTypesResult))
             {
-                throw GetMayanResponseError(result.Message);
+                ExternalResponse<QueryResponse<Models.Mayan.Document.DocumentTypeModel>> result = await documentStorageRepository.TryGetDocumentTypesAsync(ordering, page, pageSize);
+                if (result.Status != ExternalResponseStatus.Success)
+                {
+                    throw GetMayanResponseError(result.Message);
+                }
+
+                MemoryCacheEntryOptions cacheOptions = new();
+                cacheOptions.SetSlidingExpiration(TimeSpan.FromHours(2));
```

Comment:
For this change, let's make this time controllable by configuration instead of hardcoding it.